R/seroprevalence_data.R

```diff
@@ -164,13 +195,55 @@ prepare_bin_data <- function(serodata) {
 #' foi <- rep(0.02, n_years)
 #' sim_probability <- get_sim_probability(sim_data = sim_data, foi=foi)
 #' @export
-get_sim_probability <- function(sim_data, foi) {
-  sim_data <- sim_data %>%
-    mutate(birth_year = .data$tsur - .data$age_mean_f)
+get_sim_probability <- function(
+  sim_data,
+  foi,
+  mu = 0,
+  is_time_varying = TRUE,
+  is_age_varying = FALSE
+  ) {
+
+  # Checks valid model specification
+  if (is_time_varying && is_age_varying) {
+    warning(
+      "Time and age varying data simulation is not supported ",
+      "Setting `is_time_varying = FALSE`."
+    )
+    is_time_varying <- FALSE
+  }
+  if (!any(is_time_varying, is_age_varying)) {
+      warning(
+        "Type of model not specified. ",
+        "Setting `is_time_varying = TRUE`."
+        )
+      is_time_varying <- TRUE
+    }
```

Comment:
Think these would not be needed with the above suggested change to the function arguments